server/leafnode_test.go

```diff
@@ -4072,3 +4099,942 @@ func TestLeafNodeInterestPropagationDaisychain(t *testing.T) {
+			defer os.RemoveAll(sd4)
+			// deliberately pick server sA and proxy
+			confLB := createConfFile(t, []byte(fmt.Sprintf(tmplLB, sd4, hint, sA.opts.LeafNode.Port, lPort)))
+			defer removeFile(t, confLB)
+			sLB, _ := RunServerWithConfig(confLB)
+			defer sLB.Shutdown()
+
+			checkClusterFormed(t, sLA, sLB)
+
+			strmCfg := func(name, placementCluster string) *nats.StreamConfig {
+				if placementCluster == "" {
+					return &nats.StreamConfig{Name: name, Replicas: 1, Subjects: []string{name}}
+				}
+				return &nats.StreamConfig{Name: name, Replicas: 1, Subjects: []string{name},
+					Placement: &nats.Placement{Cluster: placementCluster}}
+			}
+			// Only after the system account is fully connected can streams be placed anywhere.
+			testJSFunctions := func(pass bool) {
+				ncA := natsConnect(t, fmt.Sprintf("nats://a1:a1@127.0.0.1:%d", sA.opts.Port))
+				jsA, err := ncA.JetStream()
+				require_NoError(t, err)
+				_, err = jsA.AddStream(strmCfg(fmt.Sprintf("fooA1-%t", pass), ""))
+				require_NoError(t, err)
+				_, err = jsA.AddStream(strmCfg(fmt.Sprintf("fooA2-%t", pass), "clust1"))
+				require_NoError(t, err)
+				_, err = jsA.AddStream(strmCfg(fmt.Sprintf("fooA3-%t", pass), "clustL"))
+				if pass {
+					require_NoError(t, err)
+				} else {
+					require_Error(t, err)
+					require_Contains(t, err.Error(), "insufficient resources")
+				}
+				ncL := natsConnect(t, fmt.Sprintf("nats://a1:a1@127.0.0.1:%d", sLA.opts.Port))
+				jsL, err := ncL.JetStream()
+				require_NoError(t, err)
+				_, err = jsL.AddStream(strmCfg(fmt.Sprintf("fooL1-%t", pass), ""))
+				require_NoError(t, err)
+				_, err = jsL.AddStream(strmCfg(fmt.Sprintf("fooL2-%t", pass), "clustL"))
+				require_NoError(t, err)
+				_, err = jsL.AddStream(strmCfg(fmt.Sprintf("fooL3-%t", pass), "clust1"))
+				if pass {
+					require_NoError(t, err)
+				} else {
+					require_Error(t, err)
+					require_Contains(t, err.Error(), "insufficient resources")
+				}
+			}
+			clusterLnCnt := func(expected int) error {
+				cnt := 0
+				for _, s := range c.servers {
+					cnt += s.NumLeafNodes()
+				}
+				if cnt == expected {
+					return nil
+				}
+				return fmt.Errorf("not enought leaf node connections, got %d needed %d", cnt, expected)
+			}
+
+			// Even though there are two remotes defined in tmplL, only one will be able to connect.
+			checkFor(t, 10*time.Second, time.Second/4, func() error { return clusterLnCnt(2) })
+			checkLeafNodeConnectedCount(t, sLA, 1)
+			checkLeafNodeConnectedCount(t, sLB, 1)
+			c.waitOnPeerCount(2)
+
+			if testCase.leafFunctionPreJoin {
+				cl := cluster{t: t, servers: []*Server{sLA, sLB}}
+				cl.waitOnLeader()
+				cl.waitOnPeerCount(2)
+				testJSFunctions(false)
+			} else {
+				// In cases where the leaf nodes have to wait for the system account to connect,
+				// JetStream should not be operational during that time
+				ncA := natsConnect(t, fmt.Sprintf("nats://a1:a1@127.0.0.1:%d", sLA.opts.Port))
```

Comment:
I think `defer ncA.Close()` is missing here.

Reply:
fixed